instrumentation-genai/opentelemetry-instrumentation-google-genai/tests/generate_content/test_e2e.py

```diff
@@ -0,0 +1,507 @@
+)
+def fixture_setup_yaml_pretty_formatting():
+    yaml.add_representer(_LiteralBlockScalar, _literal_block_scalar_presenter)
+
+
+def _process_string_value(string_value):
+    """Pretty-prints JSON or returns long strings as a LiteralBlockScalar"""
+    try:
+        json_data = json.loads(string_value)
+        return _LiteralBlockScalar(json.dumps(json_data, indent=2))
+    except (ValueError, TypeError):
+        if len(string_value) > 80:
+            return _LiteralBlockScalar(string_value)
+    return string_value
+
+
+def _convert_body_to_literal(data):
+    """Searches the data for body strings, attempting to pretty-print JSON"""
+    if isinstance(data, dict):
+        for key, value in data.items():
+            # Handle response body case (e.g., response.body.string)
+            if key == "body" and isinstance(value, dict) and "string" in value:
+                value["string"] = _process_string_value(value["string"])
+
+            # Handle request body case (e.g., request.body)
+            elif key == "body" and isinstance(value, str):
+                data[key] = _process_string_value(value)
+
+            else:
+                _convert_body_to_literal(value)
+
+    elif isinstance(data, list):
+        for idx, choice in enumerate(data):
+            data[idx] = _convert_body_to_literal(choice)
+
+    return data
+
+
+# Helper for enforcing GZIP compression where it was originally.
+def _ensure_gzip_single_response(data: bytes):
+    try:
+        # Attempt to decompress, first, to avoid double compression.
+        gzip.decompress(data)
+        return data
+    except gzip.BadGzipFile:
+        # It must not have been compressed in the first place.
+        return gzip.compress(data)
+
+
+# VCRPy automatically decompresses responses before saving them, but it may forget to
+# re-encode them when the data is loaded. This can create issues with decompression.
+# This is why we re-encode on load; to accurately replay what was originally sent.
+#
+# https://vcrpy.readthedocs.io/en/latest/advanced.html#decode-compressed-response
+def _ensure_casette_gzip(loaded_casette):
+    for interaction in loaded_casette["interactions"]:
+        response = interaction["response"]
+        headers = response["headers"]
+        if (
+            "content-encoding" not in headers
+            and "Content-Encoding" not in headers
+        ):
+            continue
+        if (
+            "content-encoding" in headers
+            and "gzip" not in headers["content-encoding"]
+        ):
+            continue
+        if (
+            "Content-Encoding" in headers
+            and "gzip" not in headers["Content-Encoding"]
+        ):
+            continue
+        response["body"]["string"] = _ensure_gzip_single_response(
+            response["body"]["string"].encode()
+        )
+
+
+def _maybe_ensure_casette_gzip(result):
+    if sys.version_info[0] == 3 and sys.version_info[1] == 9:
```

Comment:
Is it only broken in 3.9?

Reply:
Yes, that is correct. Version 3.9 seems to be more stringent with validating gzip content. The other versions will return the original content if it does not match the expectations of the gzip format.